File: 214/projects/07/java/NameTester.java
import java.util.Scanner;

/* NameTester.java is a 'driver' to test class Name.
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Student: Duncan Van Keulen
 * For Proj07
 * Date: 4-6-2020
 ************************************************************/


// Tests the name class
public class NameTester {
    public static void main(String[] args) {
        Name aName = new Name("John", "Paul", "Jones");
        
        assert aName.getFirst().equals("John") ;
        assert aName.getMiddle().equals("Paul") ;
        assert aName.getLast().equals("Jones") ;
        
        System.out.println(aName);
        assert aName.toString().equals("John Paul Jones");

        // Mutator tests
        aName.setFirst("Phillip");
        assert aName.getFirst().equals("Phillip");
        
        aName.setMiddle("Joseph");
        assert aName.getMiddle().equals("Joseph");

        aName.setLast("Wheelwright");
        assert aName.getLast().equals("Wheelwright");

        // read needs a scanner object
        Scanner keyboard = new Scanner(System.in);
        aName.read(keyboard);
        System.out.println(aName.toString());
        assert aName.toString().equals("Jane Renee Doe");


        System.out.println("All tests passed!");
    }
}

// Simple class used for implementing a name of First, Middle, Last
class Name {
    private String myFirst, myMiddle, myLast;

    public Name(String first, String middle, String last) {
        myFirst = first;
        myMiddle = middle;
        myLast = last;
    }

    public String getFirst(){
        return myFirst;
    }

    public String getMiddle(){
        return myMiddle;
    }

    public String getLast(){
        return myLast;
    }

    public String toString(){
        return myFirst + ' ' + myMiddle + ' ' + myLast;
    }

    public void setFirst(String first) {
        myFirst = first;
    }

    public void setMiddle(String middle) {
        myMiddle = middle;
    }

    public void setLast(String last) {
        myLast = last;
    }

    public String lfmi() {
        return myLast + ", " + myFirst + " " + myMiddle.charAt(0) + ".";
    }

    public void read(Scanner keyboard) {
        myFirst = keyboard.nextLine();
        myMiddle = keyboard.nextLine();
        myLast = keyboard.nextLine();
    }

};

File: code/projects/07/ruby/NameTester.rb
# NameTester.rb tests class Name and its operations
#
# Begun by: Dr. Adams, for CS 214 at Calvin College.
# Completed by: Duncan Van Keulen
# For Proj07
# Date: 4-6-2020
####################################################

require 'test/unit/assertions'   # needed for assert
include Test::Unit::Assertions

class Name 

   def initialize(first, middle, last)
      @first, @middle, @last = first, middle, last
   end

   # accessor methods
   attr_reader :first, :middle, :last

   # setter methods
   attr_writer :first, :middle, :last

   def fullName()
      first +  ' ' + middle + ' ' + last
   end

   def print
      puts(fullName())
      return fullName()
   end

   def lfmi
      return last + ", " + first + " " + middle[0] + ".";
   end

   # Change the values by reading them in from the keyboard
   def read
      @first = gets.chomp
      @middle = gets.chomp
      @last = gets.chomp
   end
end


def testName
   # Test the getters
   name = Name.new("John", "Paul", "Jones")

   assert name.first == "John", "first failed"
   assert name.middle == "Paul", "middle failed"
   assert name.last == "Jones", "last failed"
   assert name.fullName == "John Paul Jones", "fullName failed"
   assert name.print == "John Paul Jones", "print failed"

   # Setters and LFMI tests
   name.first = "Phillip"
   name.middle = "Joseph"
   name.last = "Wheelwright"

   assert name.first == "Phillip"
   assert name.middle == "Joseph"
   assert name.last == "Wheelwright"
   assert name.lfmi == "Wheelwright, Phillip J."

   # Read test
   name.read
   name.print
   assert name.first == "Jane"
   assert name.middle == "Renee"
   assert name.last == "Doe"
   
   print "All tests passed!\n"
end

testName

